ownCloud/Client/Actions/Actions+Extensions/OpenInAction.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  OpenInAction.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 12/11/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import ownCloudSDK
+
+class OpenInAction: Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.openin") }
+	override class var category : ActionCategory? { return .normal }
+	override class var name : String { return "Open in".localized }
+
+	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
+		if forContext.items[0].type == .collection {
+			return .none
+		}
+		return .first
+	}
+
+	private var interactionController: UIDocumentInteractionController?
+
+	override func run() {
+		guard context.items.count > 0, let viewController = context.viewController else {
+			completionHandler?(NSError(ocError: .errorInsufficientParameters))
+			return
+		}
+
+		beforeRunHandler?()
+
+		let item = context.items[0]
+
+		let controller = DownloadFileProgressHUDViewController()
+
+		OnMainThread {
```

Comment:
Maybe we should make it a convention that `run()` on actions may only be called from the main thread (and log a warning in `preRun()` if it's called on anything else, so we catch that early). You can then get rid of `OnMainThread` here.